rskj-core/src/test/java/co/rsk/blockchain/utils/BlockGenerator.java

```diff
@@ -297,8 +301,16 @@ public Block createChildBlock(Block parent, List<Transaction> txs, List<BlockHea
     }
 
     public Block createBlock(int number, int ntxs) {
-        Bloom logBloom = new Bloom();
-        Block parent = getGenesisBlock();
+        return createBlock(number, ntxs, -1);
+    }
+
+    public Block createBlock(int number, int ntxs, long gasLimit){
+        Block parent;
+        if (gasLimit == -1){
+            parent = getGenesisBlock();
+        } else {
+            parent = getGenesisBlock(gasLimit);
+        }
```

Comment:
I'd maybe change this section to this, so there's no confusion in the `-1` usage convention (and would only involve changing `1000000` to `1000000L` in the test class you modified):
```suggestion
        return createBlock(number, ntxs, null);
    }

    public Block createBlock(int number, int ntxs, Long gasLimit){
        Block parent = gasLimit == null ? getGenesisBlock() : getGenesisBlock(gasLimit);
```

Reply:
a much better and cleaner solution, thank you!

Reply:
https://github.com/rsksmart/rskj/pull/1588/commits/64c10df1c22f076a5cba5e7b6290aebcd3d74c2a